src/Select.js

```diff
@@ -117,24 +115,16 @@ export class Select extends Component {
      */
     onOutsideClick = e => {
         const { onBlur, disabled, selected } = this.props
-        const { open } = this.state
 
-        if (!open || disabled) {
+        if (disabled) {
             return
         }
 
-        const isInsideSelect = this.selectRef.current.contains(e.target)
-        const isInsideMenu =
-            this.menuRef.current && this.menuRef.current.contains(e.target)
-        const isInsideClick = isInsideSelect || (open && isInsideMenu)
-
-        if (!isInsideClick) {
-            if (onBlur) {
-                onBlur({ selected }, e)
-            }
-
-            this.handleClose()
+        if (onBlur) {
+            onBlur({ selected }, e)
         }
```

Comment:
This should be executed after `this.handleClose`, so any synchronous code called this way is not executed before the closing-logic has actually been executed.

Reply:
I changed this in 129f14b.